File: backend/albergue1/services/caracteristica_complementaria.py
from flask import Blueprint, request, jsonify, make_response
from utils.db import db

from models.caracteristica_complementaria import Caracteristica_complementaria
from schemas.caracteristica_complementaria import caracteristica_complementaria_schema, caracteristicas_complementarias_schema

car_complementarias_routes = Blueprint('car_complementarias_routes', __name__)

@car_complementarias_routes.route("/get_car_complementarias", methods=["GET"])
def get_car_complementarias():
    caracteristicas = Caracteristica_complementaria.query.all()

    if not caracteristicas:
        data = {
            'message': 'No se encontraron caracteristicas complementarias.',
            'status': 404
        }
        return make_response(jsonify(data), 404)
    else:
        result = caracteristicas_complementarias_schema.dump(caracteristicas)
        data = {
            'message': 'Caracteristicas complementarias encontradas.',
            'status': 200,
            'data': result
        }
        return make_response(jsonify(data), 200)
    
@car_complementarias_routes.route("/get_car_complementaria", methods=["POST"])
def get_car_complementaria():
    id_caracteristica = request.json.get('id_caracteristica')
    if not id_caracteristica:
        data = {
            'message': 'ID de caracteristica complementaria no proporcionado.',
            'status': 400
        }
        return make_response(jsonify(data), 400)
    else:
        raza = Caracteristica_complementaria.query.get(id_caracteristica)

        if not raza:
            data = {
                'message': 'Caracteristica complementaria no encontrada.',
                'status': 404
            }
            return make_response(jsonify(data), 404)
        else:
            result = caracteristica_complementaria_schema.dump(raza)
            data = {
                'message': 'Caracteristica complementaria encontrada.',
                'status': 200,
                'data': result
            }
            return make_response(jsonify(data), 200)

File: backend/albergue3/controllers/caracteristica_complementaria.js
const CaracteristicaComplementaria = require('../models/caracteristica_complementaria.js');

const get_car_complementarias = (req, res) => {
    CaracteristicaComplementaria.obtenerTodos((err, resultados) => {
        if (err) {
            return res.status(500).json({ mensaje: 'Error en el servidor', error: err });
        }
        if (resultados.length === 0) {
            return res.status(404).json({ mensaje: 'No se encontraron caracteristicas complementarias' });
        }
        res.status(200).json({
            message: 'Características complementarias encontradas.',
            status: 200,
            data: resultados
        });
    });
};

const get_car_complementaria = (req, res) => {
    const { id_caracteristica } = req.body;
    if (!id_caracteristica) {
        return res.status(400).json({ mensaje: 'ID de caracteristica complementaria es requerido' });
    }
    
    CaracteristicaComplementaria.obtenerPorId(id_caracteristica, (err, resultado) => {
        if (err) {
            return res.status(500).json({ mensaje: 'Error en el servidor', error: err });
        }
        if (!resultado || resultado.length === 0) {
            return res.status(404).json({ mensaje: 'Caracteristica complementaria no encontrada' });
        }
        res.status(200).json({
            message: 'Característica complementaria encontrada.',
            status: 200,
            data: resultado[0]
        });
    });
}

module.exports = {
    get_car_complementarias,
    get_car_complementaria
};
